mautic/mautic
# TPROD-170: Enabling 2-legged authentication for Mautic

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | yes
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... PENDING
| Related developer documentation PR URL | mautic/developer-documentation#... PENDING
| Issue(s) addressed                     | Fixes TPROD-170

#### Description:
This PR adds support for a 2-legged OAuth2 authentication to the Mautic API using the `client_credentials` grant.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Login as admin
3. Go to Settings > API credentials
4. Create new set of credentials
5. Generate access_token using the endpoint `oauth/v2/token` (Postman would be a good way to do this)
6. Use the following pattern in the body:
`{
    "grant_type": "client_credentials",
    "client_id": "YOUR_CLIENT_ID_FROM_STEP_4",
    "client_secret": "YOUR_CLIENT_SECRET_FROM_STEP_4"
}`
7. You should receive `access_token` in the response. Copy that.
8. Send an API request on any of the resources. For e.g: `api/assets`. In the header, send the Authorization header with the above token
9. You should receive API response without any error. 


#### BC Breaks
WIP
